internal/management/controller/instance_controller.go

```diff
@@ -171,18 +171,17 @@ func (r *InstanceReconciler) Reconcile(
 		return reconcile.Result{RequeueAfter: time.Second}, nil
 	}
 
-	restarted, err := r.reconcilePrimary(ctx, cluster)
-	if err != nil {
+	// If the instance is promoted, it will not automatically load the changed configuration files
+	// so promoted should not stop the instance reload requires
```

Comment:
```suggestion
	// Instance promotion will not automatically load the changed configuration files.
	// Therefore it should not be counted as "a restart" to prevent instance restart.
```